### PR TITLE
Action Manager | File Menu Migration

### PR DESCRIPTION
**Pull request by AMZN-daimini**
Migrates the File Menu from the legacy Action Manager to the new one.

FileMenuMigration

The migration aims to preserve the current implementation of the functionality untouched, so possible refactorings and simplifications will be considered as a follow-up.

A few notes:
- The Recent Files menu currently only updates when the File menu is opened. That is not the correct implementation, since it should be possible to show the Recent Files menu somewhere else in the interface (and that would not trigger the update). I have added a follow-up task to refactor the system so that we can notify the actions whenever the files list is changed.
- Save and Save As are shown as disabled before a level is loaded, and then enabled when a level is loaded, exactly as it happens in latest development;
- Save Level Statistics has been ported despite being always disabled in the current Editor. I have added a follow-up task to investigate and possibly remove the menu item (as that would also require changes to the python tests and possibly redcoding some of the backend);

### CODE REVIEW

**Comment by AMZN-alexpete** [comment]
wow, we really need to improve the Project Manager launch time

**Comment by AMZN-daimini** [comment]
> wow, we really need to improve the Project Manager launch time

It should be noted that I'm building on debug... but yes, perf improvements would help :D

File: Code/Editor/Core/EditorActionsHandler.h [hunk 1]
```diff
@@ -60,6 +61,11 @@ class EditorActionsHandler
     void AfterEntitySelectionChanged(
         const AzToolsFramework::EntityIdList& newlySelectedEntities, const AzToolsFramework::EntityIdList& newlyDeselectedEntities);
 
+    // Recent Files
+    bool IsRecentFileActionActive(int index);
```
Comment (cgalvan) [inline Code/Editor/Core/EditorActionsHandler.h:65]: Nitpick: Could make this `const`

**Review by cgalvan — APPROVED** [review]

File: Code/Editor/Core/EditorActionsHandler.h (hunk 1 above)
Comment (AMZN-daimini) [reply Code/Editor/Core/EditorActionsHandler.h:65]: There's always one lol. I'll add this in the follow up, thanks :D

**Review by alexmontAmazon — APPROVED** [review]

**Pull request merged**